bite-size-notes/week-3-e-optimal-control-theory.md

```diff
@@ -0,0 +1,210 @@
+
+*Statement of Maximum Principle*
+
+Consider the problem of minimizing
+$$J(u, t_f) = \int_{t_0}^{t_f} L(x, u) dt$$
+subject to $(t_f, x(t_f)) \in S = [t_0, \infty) \times S_1$ where $S_1$ is a $k$-dimensional manifold in $\mathbb{R}^n$
+$$S_1 = {x \in \mathbb{R}^n : h_1(x) = h_2(x) = \ldots = h_{n-k}(x) = 0}$$
+where $h_i$ are $C^1$ functions from $\mathbb{R}^n$ to $\mathbb{R}$ subject to
+$$\dot{x} = f(x, u), \quad x(t_0) = x_0$$
+for $u \in C[t_0, T]$ and $u(t) \in U \subset \mathbb{R}^m$ with $f$ and $L$ being $C^1$ functions.
+
+Let $u^* : [t_0, t_f] \rightarrow \mathbb{R}$ be an optimal control with state trajectory $x^* : [t_0, t_f] \rightarrow \mathbb{R}^n$ and a constant. Then
+there exists a function $p^* : [t_0, t_f] \rightarrow \mathbb{R}^n$ and a constant ${p}_0^* \leq 0$ (not both zero) for all $t \in [t_0, t_f]$ such that
+1. ${x^* }$ and ${p^* }$ satisfy Hamilton’s canonical equations,
+$$\dot{x^* } = H_p({x^* }, {u^* }, {p^* }, {p_0^* } ), \quad \dot{p^* } = -H_x({x^* } , {u^* } , {p^* } , {p_0^* })$$ with ${x^* }(t_0) = x_0$ and ${x^* }(t_f) \in S_1$ where
+$$H(x, u, p, p_0) = \langle p, f(x, u) \rangle + p_0 L(x, u)$$
+
+2. For each $t \in [t_0, t_f]$ and $u \in U$,
+$$H({x^* }, {u^* }, {p^* }, {p_0^* }) \geq H({x^* }, u, {p^* }, {p_0^* })$$
+
+3. $H({x^* }(t), {u^* }(t), {p^* }(t), {p_0^* }) = 0 \quad \forall t \in [t_0, t_f]$.
+
+
+4. The vector ${p^* }(t_f)$ is orthogonal to the tangent space of $S_1$ at ${x^* }(t_f)$. In other words,
+$$\langle {p^* }(t_f), d \rangle = 0 \quad \forall d \in T_{{x^* }(t_f)} S_1$$
+
+Equation (2) is called the maximum principle, Pontryagin’s Maximum Principle, or PMP for short. Equation (4) is a transversality condition. Equation (1.b) in the pair of Hamilton’s equations is often called the co-state or adjoint equation.
+
+This is a necessary condition for global optimality. 
+Pontryagin's maximum principle is a key concept in optimal control theory. It outlines necessary conditions for determining the optimal control strategy for a dynamical system, especially when there are constraints involved. This principle asserts that to find the best control along with the optimal state trajectory, one must solve the Hamiltonian system—a two-point boundary value problem—and satisfy the maximum condition of the control Hamiltonian. These conditions are crucial for identifying the optimal control strategy, particularly under certain convexity assumptions on the objective and constraint functions.
+
+#### Principle of Optimality 
+
+The principle of optimality is a fundamental concept in optimal control theory that guides the search for optimal control strategies. At its core, it asserts that an optimal control strategy for a complex system can be decomposed into optimal control decisions at each smaller time interval.
+
+The principle of Optimality states that:
+
+For every $(t,x) \in [t_0,t_1) \times \mathbb{R}$ and every $\Delta t \in (0, t_1 - t]$, the value function $V$ satisfies the relation
```

Comment:
It's kind confusing to me. Does the [t_0,t_1) mean that t0 included and t1 is not included. Also, in the following equations and time intervals, sometimes you used [], sometimes [) and sometimes (]. Is that different or same? or make any difference for the equations you gave?

Reply:
Yes, your understanding is correct. The [t_0,t_1) means that t_0 is included and t_1 is not. It is referred to as a closed interval on t_0 and an open interval on t_1. 
In the following equation, the time interval is open on 0 and closed on t_1 - t, which implies that 0 is not included which t_1 - t is included. 

the [], [), and (] are different from each other. 

In many cases, using different types of intervals doesn't affect the equations themselves but rather clarifies the boundary conditions or the range of values being considered, but in some specific cases, the choice of interval notation may make a difference, especially when dealing with boundary conditions or continuity concerns.